backend/data_preparation/crawler/fire_crawler.py

```diff
@@ -0,0 +1,318 @@
+    pass
+
+
+class FireEvent:
+    # the class to represent a FireEvent object, a fire event is a link on the rmgsc website
+    def __init__(self, year: int, state: str, url_name: str, fire_id=-1):
+        """
+        Takes year, state, url_name and make a new FireEvent object
+        :param year: the year when the fire event occurred.
+                e.g. 2017
+        :param state: the state where the fire event occurred.
+                e.g. 'California'
+        :param url_name: the name of this fire event on rmgsc website, should contains underscore.
+                e.g. "Happy_Camp_Mountain"
+        :param fire_id: the id of the fire event in database, -1 for new fire events that is not inserted into the
+                database
+        """
+        self.fire_id = fire_id
+        self.year = year
+        self.state = state
+        self.url_name = url_name
+
+    def __eq__(self, other):
+        return self.year == other.year and self.state == other.state and self.url_name == other.url_name
+
+    def __hash__(self):
+        return hash(self.to_url())
+
+    @classmethod
+    def from_tuple(cls, tuple_for_information: Tuple[Any]) -> 'FireEvent':
+        """
+        Generates a FireEvent object from a tuple of information.
+        :param tuple_for_information: Tuple.
+               e.g. (2015, 'California', 'Happy_Camp'), (111, 2017, 'California', 'Eclipse')
+        :return: FireEvent object.
+               e.g. FireEvent(-1, 2015, 'California', 'Happy_Camp'), FireEvent(111, 2017, 'California', 'Eclipse')
+        """
+        if len(tuple_for_information) == 3:
+            return cls(tuple_for_information[0], tuple_for_information[1], tuple_for_information[2])
+        else:
+            return cls(tuple_for_information[1], tuple_for_information[2], tuple_for_information[3],
+                       tuple_for_information[0])
+
+    def is_valid(self) -> bool:
+        """
+        Judges if this fire event is valid. A valid fire event should have reasonable fire name.
+        :return: bool
+        """
+        # to delete exceptions: "[To Parent Directory]" "whatever.zip"
+        # ActivePerim is an useless link which contains metadata over the year, existing in links before 2016 Skip it
+        return "." in self.url_name or "[To Parent Directory]" == self.url_name or self.url_name == "ActivePerim"
+
+    def __str__(self) -> str:
+        """
+        Transforms a FireEvent into a string.
+        :return: str
+        """
+        if id != -1:
+            return f"Fire Event {self.fire_id}: {self.url_name} in year {self.year}, state {self.state}"
+        return f"Fire Event Unknown: {self.url_name} in year {self.year}, state {self.state}"
+
+    def to_url(self) -> str:
+        """
+        Converts a FireEvent object into the url
+        :return: url. e.g. "https://rmgsc.cr.usgs.gov/outgoing/GeoMAC/2017_fire_data/California/Eclipse"
+        """
+        current_year = datetime.datetime.now().date().year
```

Comment:
this `now()` might behavior differently on different environment. it usually gets the server's time zone. I wonder if it will cause time mismatch problems here. (it only uses year part, I guess it won't be a big issue? just want to double check)